content/docs/1.10.0/advanced-resources/rebuilding/_index.md

```diff
@@ -1,4 +1,114 @@
+        - If **NO**, directly transfer all data from the source replica to the destination replica. [Full Replica Rebuilding](#full-replica-rebuilding)
+        - If **YES**, check whether the value of snapshot checksum files between the target replica and the source replica exists and is identical:
+            - If **YES**, Longhorn skips the corresponding snapshot data transfer, which reduces the CPU usage, disk IO, network IO, and rebuilding time. [Fast Replica Rebuilding](#fast-replica-rebuilding)
+            - If **NO**, Longhorn calculates and compares block-by-block checksums for the snapshot file between the target replica and the source replica. If there is a mismatch, the corresponding data block will be synced. [Delta Replica Rebuilding](#delta-replica-rebuilding)
+
+## Full Replica Rebuilding
+
+If the replica is unrecoverable or replenished, Longhorn will synchronize the whole data from the healthy replica. Reconstruct the replica by sending the full snapshot chain.
+
+Full replica rebuilding will waste network bandwidth and generate numerous disk WRITE operations on the target replica node, but it is the only way to rebuild a replica if there is no data on the target node.
+
+## Delta Replica Rebuilding
+
+6. Compare block-by-block checksums if [Fast Replica Rebuilding](#fast-replica-rebuilding) is not enabled. Longhorn engine will compare each snapshot files block-by-block between the target replica and the source replica. The checksum is computed using SHA-512 method.
+7. If local and remote checksums do not match, receiver server of the target replica will synchronize missing or mismatched data.
+
+- **Pros**:
+  - Reduce network bandwidth consumption.
+  - Reduce disk IO.
+- **Cons**:
+  - Increased CPU overhead.
+  - Rebuilding time is influenced by CPU performance.
+
+## Fast Replica Rebuilding
+
+Fast rebuilding is enabled when:
+
+- The fast replica building setting is enabled:
+  - `fast-replica-rebuild-enabled: true`
+- Snapshot checksum files are created:
+  - `snapshot-data-integrity`: `enabled` or
+  - `snapshot-data-integrity-immediate-check-after-snapshot-creation: true`
+
+6. Skip checking unchanged snapshots if the checksums are the same in the checksum files of snapshots.
+7. If the value of snapshot checksum files between the target replica and the source replica does not exist or is not identical, it will go to the delta rebuilding flow.
+
+- **Pros**:
+  - Minimize network bandwidth consumption.
+  - Minimize disk IO.
+- **Cons**:
+  - Calculating the snapshot checksum can be time-consuming.
+
+The snapshot checksum is computed:
+
+- Immediately after the snapshot is created with the `snapshot-data-integrity-immediate-check-after-snapshot-creation` setting enabled.
+- Regularly (by default: 7 days) if the `snapshot-data-integrity` setting is `enabled`.
+
+For more details, [Fast Replica Rebuilding](./fast-replica-rebuilding).
+
+## Factors That Affect Rebuilding Performance
```

Comment:
Besides, I am not sure if we can mention the concurrent rebuilding setting in this section

Reply:
Removed. Thanks